dask/highlevelgraph.py

```diff
@@ -57,12 +57,40 @@ class Layer(collections.abc.Mapping):
     """
 
     annotations: Optional[Mapping[str, Any]]
+    collection_annotations: Optional[Mapping[str, Any]]
 
-    def __init__(self, annotations: Mapping[str, Any] = None):
+    def __init__(
+        self,
+        annotations: Mapping[str, Any] = None,
+        collection_annotations: Mapping[str, Any] = {},
```

Comment:
It would be good to avoid a mutable default for a keyword argument. What you have here is totally fine today, but can lead to unexpected behavior in the future which is hard to track down. 

```suggestion
        collection_annotations: Mapping[str, Any] = None,
```